### ISSUE TITLE
🐛 BUG: Build fails when referencing "public" javascript files.

### ISSUE THREAD

**Issue by johnhenry** [snowpackjs/astro#757]
### What package manager are you using?

npm

### What operating system are you using?

Mac

### Describe the Bug
(Astro version 0.19.1)

Build fails when importing a javascript file from "./public" directory.

Note: this works properly on the dev server -- just not when building for production.
Note: importing CSS files in the same way does not cause a similar issue.

### Steps to Reproduce

1. Init generic astro project
2. Run `npm install`
3. Add basic javascript file at /public/index.js
4. Add basic astro html file at /src/pages/indes.astro that references the above javascript file.
5. Run `npm run build`
6. Error! Build fails with message 'Error: file:///.../src/pages/index.astro: could not find "index.js"
   at file:///.../node_modules/astro/src/build.ts:140:25'
7. Run `npm start`
8. Visit http://localhost:3000/
9. Open console. Note the javascript file has bee pulled in to /src/pages/index.astro.

### Link to Minimal Reproducible Example (Optional)

https://github.com/johnhenry/astro-build-fail-imports

**Comment by FredKSchott** [comment]
This wouldn't be expected to work, since we don't rewrite any references inside of your HTML template (same reason why img src="/"` needs to be an absolute URL).

if you change that to `import '/index.js'` does that work?

See related issue: https://github.com/snowpackjs/astro/issues/310

**Comment by johnhenry** [comment]
@FredKSchott thanks for the reply.

I may be confused... I'm specifically trying to get astro to NOT attempt to resolve the URL for JavaScript files during the build process:
  - in the same way that it does not attempt a resolution for CSS
  - in the same way that it does not attempt a resolution when running the dev server
 
 From the include example, this code, which imports a js file, causes `astro build` to fail.
 
 ```html
  <script type="module">
      import "./index.js";
  </script> 
 ```
 
  Whereas this code, which imports a css file builds.
  
 ```html
 <style>
      @import "./index.css";
 </style>
 ```
 
At best, this seems inconsistent?
 
Further, this only seems to cause issues when running `astro build`, whereas the dev server works properly.

Based on my understanding of the docs [here](https://docs.astro.build/core-concepts/astro-components#url-resolution), imports within the frontmatter section of an .astro file should be resolved during the build process,
but URLs -- as strings within the HTML template -- should blindly written out to be resolved at run time. Is my understanding of this incorrect?

I do not think that this issue has anything to do with rewriting references -- the issue seems to be that for some `astro build` attempts to process some (specifically javascript) files if referenced.

**Comment by matthewp** [comment]
Thanks! This does seem like a bug then.

**Comment by johnhenry** [comment]
As temporary workaround, wrapping the full tag in a string works.

```javascript
{`<script type="module">
     import "./index.js";
 </script> `}
```

**Comment by johnhenry** [comment]
I do not think that these lines produces the desired behavior and I believe simply removing them will fix this issue: https://github.com/snowpackjs/astro/blob/main/packages/astro/src/build.ts#L148-L168.

They appear to be at somewhat connected to an attempt to properly [rewrite urls](https://github.com/snowpackjs/astro/issues/104), but as @FredKSchott points out, we shouldn't be writing references. Further, I think the issues are addressed by the recently introduced `Astro.resolve`.

**Comment by johnhenry** [comment]
Or, perhaps a better, more targeted fix would be to return a warning rather than throw an error on these lines https://github.com/snowpackjs/astro/blob/main/packages/astro/src/build.ts#L153-L159 ?

**Comment by mascarell** [comment]
Hello!

Was this fixed @johnhenry ? Found myself having this problem too

**Comment by FredKSchott** [comment]
Thanks for the report! We believe that this has been fixed in astro v0.21, going out today. Let us know if you can still reproduce the issue after updating to v0.21 and we'll be happy to take a look.
